tests/decap/test_decap.py

```diff
@@ -0,0 +1,242 @@
+    topology = testbed["topo"]["properties"]["topology"]
+
+    # uplink ports
+    uplink_ports_str = " ".join(get_uplink_ports(topology, topo_type))
+
+    # downlink ports
+    downlink_ports_str = " ".join(get_downlink_ports(topology, topo_type))
+
+    fibs = []
+
+    podset_number = 5  # Limit the number of podsets to limit test execution time
+    tor_number = 16
+    tor_subnet_number = 2
+
+    # routes to uplink
+    routes_uplink_v4 = []
+    routes_uplink_v6 = []
+    if topo_type == "t0":
+        routes_uplink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "")
+        routes_uplink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "")
+    elif topo_type == "t1":
+        routes_uplink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="spine")
+        routes_uplink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="spine")
+
+    for prefix, _, _ in routes_uplink_v4:
+        fibs.append("{} {}".format(prefix, uplink_ports_str))
+    for prefix, _, _ in routes_uplink_v6:
+        fibs.append("{} {}".format(prefix, uplink_ports_str))
+
+    routes_downlink_v4 = []
+    routes_downlink_v6 = []
+    if topo_type == "t1":
+        routes_downlink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="tor")
+        routes_downlink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
+                                            0, 0, 0,
+                                            "", "", router_type="tor")
+
+    for prefix, _, _ in routes_downlink_v4:
+        fibs.append("{} {}".format(prefix, downlink_ports_str))
+    for prefix, _, _ in routes_downlink_v6:
+        fibs.append("{} {}".format(prefix, downlink_ports_str))
+
+    ptfhost.copy(content="\n".join(fibs), dest="/root/fib_info.txt")
+
+
+def prepare_ptf(ptfhost, testbed, cfg_facts):
+    logger.info("Remove IP and change MAC on PTF container")
+    ptfhost.script("./scripts/remove_ip.sh")
+    ptfhost.script("./scripts/change_mac.sh")
+
+    logger.info("Copy PTF scripts to PTF container")
+    ptfhost.copy(src="ptftests", dest="/root")
+
+    gen_fib_info(ptfhost, testbed, cfg_facts)
+
+
+@pytest.fixture(scope="module")
+def setup_teardown(request, testbed, duthost, ptfhost):
+
+    # Initialize parameters
+    dscp_mode = "pipe"
+    ecn_mode = "standard"
```

Comment:
is 'pipe' + 'copy_from_outer' more prevailing? If so, why not set that as default and remove hwsku check for Broadcom and Marvell?

My concern is that any hwsku not in broadcom/mellanox/marvell would default to 'pipe' + 'standard', is this desirable?

Reply:
Sure, I'll change the default to `pipe` and `copy_from_outer`. I didn't know that `copy_from_outer` is more prevailing. Probably I was deceived by the word 'standard'.